MediaBrowser.Providers/MediaInfo/AudioResolver.cs

```diff
@@ -0,0 +1,164 @@
+#pragma warning disable CA1002, CS1591
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Emby.Naming.Audio;
+using Emby.Naming.Common;
+using Jellyfin.Extensions;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.MediaEncoding;
+using MediaBrowser.Controller.Providers;
+using MediaBrowser.Model.Dlna;
+using MediaBrowser.Model.Dto;
+using MediaBrowser.Model.Entities;
+using MediaBrowser.Model.Globalization;
+using MediaBrowser.Model.MediaInfo;
+
+namespace MediaBrowser.Providers.MediaInfo
+{
+    public class AudioResolver
+    {
+        public async Task<List<MediaStream>> GetExternalAudioStreams(
+            Video video,
+            int startIndex,
+            IDirectoryService directoryService,
+            NamingOptions namingOptions,
```

Comment:
Instead of having so many deps as arguments to the method, you should make a constructor. Just don't `new` the class whenever you use it :)

The deps in question are:
1. NamingOptions
2. ILocalizationManager
3. IMediaEncoder

Reply:
With not newing the class you mean the three arguments mentioned, correct? Or do you mean the AudioResolver? :)

Reply:
I mean `AudioResolver`. It should only be newed once if possible.